src/dev/flang/fuir/analysis/AbstractInterpreter.java

```diff
@@ -63,8 +65,51 @@ public class AbstractInterpreter<VALUE, RESULT> extends ANY
    * Interface that defines the operations of the actual interpreter
    * that processes this code.
    */
-  public static abstract class ProcessStatement<VALUE, RESULT> extends ANY
+  public static abstract class ProcessStatement<VALUE, RESULT> extends ANY implements HasSourcePosition
   {
+
+    /**
+     * FUIR we are currently interpreting
+     */
+    FUIR _fuir;
```

Comment:
What is the benefit of having a _fuir field here? Is it not possible to use the final _fuir field of AbstractInterpreter?

Reply:
The `Processor` currently is a static class, so it cannot access `_fuir` from the surrounding class. It might, however, make sense to change this.